.eslintrc.js

```diff
@@ -2,4 +2,18 @@ module.exports = {
   extends: ['@alhadis/eslint-config'],
   parser: '@typescript-eslint/parser',
   plugins: ['@typescript-eslint'],
+  rules: {
+    "consistent-this": 0,
+    "prefer-rest-params": 0
+  },
+  overrides: [
+    {
+      "files": [
+        "test/*"
+      ],
+      "globals": {
+        "expect": true
+      }
+    }
+  ]
```

Comment:
Just curious, what text-editor do you use? EditorConfig support should've formatted that appropriately… 🤔

Reply:
I didn't added a proper extension yet. I'm using VSCode.

Reply:
I thought as much. VSCode claims to have [EditorConfig support built-in](https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2017-relnotes-v15.0#coding-convention-support-through-editorconfig), it just needs configuration.

Reply:
I think this page describes Visual Studio features. Visual Studio Code is a lightweight version of Visual Studio and I need to install an extension

Reply:
Btw I just installed it.

Reply:
OH. Uh, okay, that makes things a bit more interesting. 🤣 I've never used Visual Studio, as you can probably tell (I'm from Unix-land 😉).

You might want to [check this out instead then](https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2017-mac-relnotes-v7.5#release-highlights).

Reply:
> OH. Uh, okay, that makes things a bit more interesting. rofl I've never used Visual Studio, as you can probably tell (I'm from Unix-land wink).
Yes, it looks like you are a system programmer (C and/or C++ probably) because each and every JS project repo you own contains Makefile =)

Reply:
You may be surprised, but VSCode is built using Electron (https://www.electronjs.org/) which is based on Node.js

Reply:
> Yes, it looks like you are a system programmer

HAH! Nope, wrong. 😉 I've been working with web technologies since... eh, 2004-2005, and most of my projects are actually pretty high-level stuff. C/C++'s pointer syntax scares the shit out of me. 😁

Nope, I use Makefiles because they're they're clean and declarative, and encourage you to leverage existing system commands (of which there are plenty) instead of downloading a crapload of tools needed to execute an overly-complex workflow. I prefer that complexity stay in the codebase I'm trying to build. 😉

FYI, you can run Make and many other Unix tools on Windows by installing [GOW](https://github.com/bmatzelle/gow/wiki); and you'll only need to do so once, as it's a very unintuitive installation.

Reply:
> HAH! Nope, wrong. wink I've been working with web technologies since... eh, 2004-2005, and most of my projects are actually pretty high-level stuff. C/C++'s pointer syntax scares the shit out of me. grin

I just started to learn C++, it's useful for 2 reasons:
 - Node.js C++ bindings
 - WASM (Using existing C++ codebase you can make a WASM file and run in browser environment, see: https://emscripten.org/)

> Nope, I use Makefiles because they're they're clean and declarative, and encourage you to leverage existing system commands (of which there are plenty) instead of downloading a crapload of tools needed to execute an overly-complex workflow. I prefer that complexity stay in the codebase I'm trying to build. wink

If you already using NPM for your project, then it might be easier to just add a scripts in package.json

> FYI, you can run Make and many other Unix tools on Windows by installing [GOW](https://github.com/bmatzelle/gow/wiki); and you'll only need to do so once, as it's a very unintuitive installation.

I don't use Windows, because it's crap IMHO, I'm okay with Ubuntu Linux for now

Reply:
Wait, Visual Studio runs on Linux? 🤔 The more you know…

> If you already using NPM for your project

I'm not using NPM by choice. In fact, it's pretty much only used for installing/updating ESLint, C8, and Mocha/Chai for linting and running tests. And I'm trying to move away from the latter two.

> I just started to learn C++

I'd learn Rust instead. 😉 It's especially good at working with WebAssembly.

Reply:
> Wait, Visual Studio runs on Linux? thinking The more you know…

Yes, from the Electron link above you may find that it's cross-platform

Reply:
> It's not that hard to bundle Node.js + Chrome browser + Electron itself

It is on BSD systems. 😜 Shit, it took **years** to port Electron to OpenBSD…

Reply:
> It is on BSD systems. stuck_out_tongue_winking_eye Shit, it took **years** to port Electron to OpenBSD…

Here is a link on build (last message): https://github.com/marktext/marktext/issues/2655#issuecomment-1003303073

Reply:
LOL, yeah, we pretty much cribbed FreeBSD's port and waited on _them_ to do all the hard work… 😁

Reply:
The version is quite old (8.2.0) there, while current stable release is [Electron 20.0.0](https://www.electronjs.org/releases/stable#20.0.0)

Reply:
Yep. That'd be because it's a nightmare to build, and IIRC, that version was only successfully ported by building against the Chromium port (as opposed to Electron's own version of Chrome).

I can't blame anybody for putting off wanting to update that pigsty of a codebase. 😫